src/System.IO.Compression.Brotli/System/IO/Compression/Brotli/BrotliPrimitives.cs

```diff
@@ -16,41 +17,67 @@ public struct BrotliPrimitives
         const int defQuality = 11;
         const int defLgWin = 24;
 
-        public static bool Compress(ReadOnlySpan<byte> source, Span<byte> destination, out nuint bytesConsumed, out nuint bytesWritten)
+        public static bool Compress(ReadOnlySpan<byte> source, Span<byte> destination, out int bytesConsumed, out int bytesWritten)
         {
             return Compress(source, destination, out bytesConsumed, out bytesWritten, defQuality, defLgWin);
         }
+        public static bool Compress(ReadOnlySpan<byte> source, Span<byte> destination, out int bytesConsumed, out int bytesWritten, int quality, int lgwin)
+        {
+            return Compress(source, destination, out bytesConsumed, out bytesWritten, quality, lgwin, BrotliNative.BrotliEncoderMode.Generic);
+        }
+
+        public static bool Compress(ReadOnlySpan<byte> source, Span<byte> destination, out int bytesConsumed, out int bytesWritten, BrotliNative.BrotliEncoderMode encMode)
+        {
+            return Compress(source, destination, out bytesConsumed, out bytesWritten, defQuality, defLgWin, encMode);
+        }
 
-        public static bool Compress(ReadOnlySpan<byte> source, Span<byte> destination, out nuint bytesConsumed, out nuint bytesWritten, int quality, int lgwin)
+        public static bool Compress(ReadOnlySpan<byte> source, Span<byte> destination, out int bytesConsumed, out int bytesWritten, int quality, int lgwin, BrotliNative.BrotliEncoderMode encMode)
         {
+            if (quality > defQuality || quality <= 0) throw new System.ArgumentOutOfRangeException(BrotliEx.WrongQuality);
+            if (lgwin > defLgWin || lgwin <= 0) throw new System.ArgumentOutOfRangeException(BrotliEx.WrongWindowSize);
             unsafe
             {
-                bytesConsumed = (nuint)source.Length;
-                bytesWritten = (nuint)0;
+                bytesConsumed = (int)source.Length;
+                bytesWritten = (int)0;
                 IntPtr bufIn, bufOut;
                 fixed (byte* inBytes = &source.DangerousGetPinnableReference())
                 fixed (byte* outBytes = &destination.DangerousGetPinnableReference())
                 {
                     bufIn = new IntPtr(inBytes);
                     bufOut = new IntPtr(outBytes);
-                    return BrotliNative.BrotliEncoderCompress(quality, lgwin, BrotliNative.BrotliEncoderMode.Generic, bytesConsumed, bufIn, ref bytesWritten, bufOut);
+                    nuint written = 0,consumed=(nuint)bytesConsumed;
```

Comment:
spacing and I would split these into separate lines.
Also, why not set consumed to source.Length instead or even get rid of the consumed variable altogether?